crates/bevy_pbr/src/render/gpu_preprocess.rs

```diff
@@ -57,20 +64,44 @@ pub struct GpuMeshPreprocessPlugin {
 
 /// The render node for the mesh uniform building pass.
 pub struct GpuPreprocessNode {
-    view_query: QueryState<(Entity, Read<PreprocessBindGroup>)>,
+    view_query: QueryState<(
+        Entity,
+        Read<PreprocessBindGroup>,
+        Read<ViewUniformOffset>,
+        Has<GpuCulling>,
+    )>,
 }
 
-/// The compute shader pipeline for the mesh uniform building pass.
+/// The compute shader pipelines for the mesh uniform building pass.
 #[derive(Resource)]
+pub struct PreprocessPipelines {
+    /// The pipeline used for CPU culling. This pipeline doesn't populate
+    /// indirect parameters.
+    pub direct: PreprocessPipeline,
+    /// The pipeline used for GPU culling. This pipeline populates indirect
+    /// parameters.
+    pub gpu_culling: PreprocessPipeline,
```

Comment:
Is there a reason to not call this indirect? I'm guessing it will be because of the upcoming combinations you have mentioned elsewhere.

Reply:
There will probably be GPU occlusion culling too, which can be toggled on and off as it has a performance cost. So it's reasonable to support GPU frustum culling but not GPU occlusion culling.